FashionConnect_ui/src/components/Main/Main.jsx

```diff
@@ -19,12 +22,73 @@ export default function Main({ collections, setCollections }) {
       throw error;
     }
   };
+  
+
+  useEffect(() => {
+    fetchNewProducts();
+  }, []);
+
+  const fetchNewProducts = async () => {
+    try {
+      const response = await fetch(
+        `https://apidojo-hm-hennes-mauritz-v1.p.rapidapi.com/products/list?country=us&lang=en&currentpage=${currentPage}&pagesize=${pageSize}`,
+        {
+          headers: {
+            'X-RapidAPI-Key': '4ecf7b02f6mshe1fb9d190d9f5f0p1dc11cjsn12e4e06a025c',
+            'X-RapidAPI-Host': 'apidojo-hm-hennes-mauritz-v1.p.rapidapi.com',
+          },
+        }
+      );
+      const data = await response.json();
+
+      if (data.results.length === 0) {
+        // No more products to fetch, stop pagination
+        return;
+      }
+
+      // Map the API response to match the database schema
+      const mappedProducts = data.results.map((result) => ({
+
+        Name: result.name,
+        Price: result.price?.value || 0, 
+        Image_URL: result.images[0]?.url || null, 
+        CategoryName: result.categoryName,
+        BrandName: result.brandName,
+      }));
+
+      setProducts((prevProducts) => [...prevProducts, ...mappedProducts]);
+      setCurrentPage((prevPage) => prevPage + 1);
+
+      // Send the new products to the backend for storage
+      await sendNewProductsToBackend(mappedProducts);
+    } catch (error) {
+      throw error;
```

Comment:
Usually we want the catch in try/catch to do something with the error, like log them or have special handling.